### PR TITLE
Add a public and private key generation and some routing fixes

### PR DESCRIPTION
**Pull request by ndufreche**
There is large cleaning in the routing (missing parameters principally)

Add 3 new methods in the user entity.

Add REST API key generation during the user creation.

### CODE REVIEW

File: Security/User.php [hunk 1]
```diff
@@ -585,4 +585,34 @@ public function setState($state)
...
+
+    /**
+     * Generate a random Api pulbic and private key
+     * @return void
+     */
+    public function generateRandomApiKey()
+    {
+        $this->_api_key_private = md5($this->id . uniqid());
+
+        $this->_api_key_public = $this->generateApiPublicKey();
```
Comment (mickaelandrieu) [inline Security/User.php]: can you return ``$this`` ?

Comment (ndufreche) [inline Security/User.php]: Oh, yes sure.

**Comment by mickaelandrieu** [comment]
:+1:

**Comment by eric-chau** [comment]
:+1:

**Comment by mickaelandrieu** [comment]
thank you @ndufreche

**Pull request merged**